### DISCUSSION TITLE
Read AOIs

### DISCUSSION THREAD

**Discussion by NinoG23** [dmroeder/pylogix#253] [Q&A]
I am wanting to scan my network to see what AOI's are being used in a number of AB PLCs and the AOI's revision. Is there a method to complete this??

**Comment by dmroeder** [comment]
Sorry for the delay @NinoG23.  The short answer is no.

When the tag list is requested by pylogix, the PLC also returns AOI's, but I filter them out.  Mainly because they aren't part of the tag list.

Let's say I modified pylogix to save the AOI names.  I don't know of a way to get the revision, nor do I know how to tell if it is actually used anywhere.  The closest I can think of is that there is a tag that is of your AOI type, that doesn't mean it is used in the logic, just that the tag exists.

**Comment by TheFern2** [comment]
Dunno about you but you should always know what software is on your PLCs, if you don't you are not controlling your software at all and that's dangerous in controls systems, just my 2¢.
